databricks/koalas/frame.py

```diff
@@ -4008,6 +4008,66 @@ def duplicated(self, subset=None, keep="first") -> "Series":
             )
         )
 
+    # TODO: support other as DataFrame or array-like
+    def dot(self, other: "Series") -> "Series":
+        """
+        Compute the matrix multiplication between the DataFrame and other.
+
+        This method computes the matrix product between the DataFrame and the
+        values of an other Series
+
+        It can also be called using ``self @ other`` in Python >= 3.5.
```

Comment:
We need to define `__matmul__` for this as same as `Series`?

https://github.com/databricks/koalas/blob/026262dcd7e455be135b7c1e0971768c3c9d37db/databricks/koalas/series.py#L4820-L4824

Reply:
Sounds good!